x-pack/plugins/security/server/config.ts

```diff
@@ -147,8 +160,8 @@ export const ConfigSchema = schema.object({
     schema.string({ minLength: 32, defaultValue: 'a'.repeat(32) })
   ),
   session: schema.object({
-    idleTimeout: schema.nullable(schema.duration()),
-    lifespan: schema.nullable(schema.duration()),
+    idleTimeout: schema.maybe(schema.oneOf([schema.duration(), schema.literal(null)])),
```

Comment:
__note:__ `nullable` doesn't let us to distinguish default value from the value that was explicitly set to `null`, just making this consistent with the provider-scoped settings. This change is required for two pending features we have (sensible defaults for these values in 8.0 and anonymous access).

Reply:
I think that's a great idea! I agree `null` is confusing and not an universal thing, `0` feels much more intuitive. We actually support 0, but I'd define our current behavior as an old bug (session expires immediately), I'll fix it.

Reply:
Done in https://github.com/elastic/kibana/pull/82583/commits/a7a1c35ef0ddeb0a38f21f99c448efcd967e953c